cmd/ops_agent_uap_plugin/service.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+
+	pb "github.com/GoogleCloudPlatform/ops-agent/cmd/ops_agent_uap_plugin/google_guest_agent/plugin"
+)
+
+// PluginServer implements the plugin RPC server interface.
+type OpsAgentPluginServer struct {
+	pb.UnimplementedGuestAgentPluginServer
+	server *grpc.Server
+	// cancel is the cancel function to be called when core plugin is stopped.
+	cancel context.CancelFunc
+}
+
+// Apply applies the config sent or performs the work defined in the message.
+// ApplyRequest is opaque to the agent and is expected to be well known contract
+// between Plugin and the server itself. For e.g. service might want to update
+// plugin config to enable/disable feature here plugins can react to such requests.
+func (ps *OpsAgentPluginServer) Apply(ctx context.Context, msg *pb.ApplyRequest) (*pb.ApplyResponse, error) {
+	return &pb.ApplyResponse{}, nil
+}
+
+// Start starts the plugin and initiates the plugin functionality.
+// Until plugin receives Start request plugin is expected to be not functioning
+// and just listening on the address handed off waiting for the request.
+func (ps *OpsAgentPluginServer) Start(ctx context.Context, msg *pb.StartRequest) (*pb.StartResponse, error) {
+	if ps.cancel != nil {
+		log.Printf("The Ops Agent plugin is started already, skipping the current request")
+		return nil, status.Errorf(1, "Ops Agent Plugin is started already")
+	}
```

Comment:
Does UAP define what the plugin should do if a plugin that's already started receives another `Start` signal? In particular, should it return an error or should it only return an grpc error if there's a failure in the startup process?

(The same question applies to `Stop`)

Reply:
Thanks for pointing out.  UAP team says that ideally we don't need to worry about handling duplicate requests, but I am keeping the check just in case. 

I have updated the code to only log the duplicate requests, instead of throwing errors as it'd trigger retries.